### ISSUE TITLE
Export ResultSet

### ISSUE THREAD

**Issue by alextakitani** [serge-rider/dbeaver#1138]
I'm using DBeaver 3.8.3 on Linux Mint 18.1

When I try to export a resultset to csv from a PostgreSQL query, it doesn't work and the error message is:

SQL Error [25P02]: ERROR: current transaction is aborted, commands ignored until end of transaction block

The file is not generated.

**Comment by serge-rider** [comment]
Can't reproduce. 
This error (current transaction is aborted .. ) happens in PostgreSQL AFTER some error occurred. In transactional mode all consequent queries will fail till explicit commit/rollback.
So there must be some real error which broke export.

Please check Error Log view or error log (https://github.com/serge-rider/dbeaver/wiki/Log-files).

**Comment by alextakitani** [comment]
Hello! Thanks for your response.

Checked the logs, the problem is when DBeaver tries to do a rowcount:

2017-01-11 12:36:07.001 - Can't retrieve row count from 'SELECT "orders".* FROM "orders" WHERE (extract(month from created_at) = 3 and extract(year from created_at) = 2016)  ORDER BY "orders"."created_at" desc'
org.jkiss.dbeaver.model.exec.DBCException: Error executing row count
	at org.jkiss.dbeaver.ui.editors.sql.SQLEditor$QueryResultsContainer.countData(SQLEditor.java:1731)
	at org.jkiss.dbeaver.tools.transfer.database.DatabaseTransferProducer.transferData(DatabaseTransferProducer.java:99)
	at org.jkiss.dbeaver.tools.transfer.database.DatabaseTransferProducer.transferData(DatabaseTransferProducer.java:1)
	at org.jkiss.dbeaver.tools.transfer.wizard.DataTransferJob.transferData(DataTransferJob.java:96)
	at org.jkiss.dbeaver.tools.transfer.wizard.DataTransferJob.run(DataTransferJob.java:65)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:103)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)
Caused by: org.jkiss.dbeaver.model.exec.DBCException: SQL Error [42803]: ERROR: column "orders.created_at" must appear in the GROUP BY clause or be used in an aggregate function
  Position: 124
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCPreparedStatementImpl.executeStatement(JDBCPreparedStatementImpl.java:205)
	at org.jkiss.dbeaver.ui.editors.sql.SQLEditor$QueryResultsContainer.countData(SQLEditor.java:1713)
	... 6 more
Caused by: org.postgresql.util.PSQLException: ERROR: column "orders.created_at" must appear in the GROUP BY clause or be used in an aggregate function
  Position: 124
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2284)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2003)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:200)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:424)
	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:161)
	at org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:155)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCPreparedStatementImpl.execute(JDBCPreparedStatementImpl.java:258)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCPreparedStatementImpl.executeStatement(JDBCPreparedStatementImpl.java:202)
	... 7 more
2017-01-11 12:36:07.135 - null
org.jkiss.dbeaver.model.exec.DBCException: SQL Error [25P02]: ERROR: current transaction is aborted, commands ignored until end of transaction block
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:141)
	at org.jkiss.dbeaver.runtime.sql.SQLQueryJob.executeSingleQuery(SQLQueryJob.java:344)
	at org.jkiss.dbeaver.runtime.sql.SQLQueryJob.extractData(SQLQueryJob.java:708)
	at org.jkiss.dbeaver.ui.editors.sql.SQLEditor$QueryResultsContainer.readData(SQLEditor.java:1682)
	at org.jkiss.dbeaver.tools.transfer.database.DatabaseTransferProducer.transferData(DatabaseTransferProducer.java:113)
	at org.jkiss.dbeaver.tools.transfer.database.DatabaseTransferProducer.transferData(DatabaseTransferProducer.java:1)
	at org.jkiss.dbeaver.tools.transfer.wizard.DataTransferJob.transferData(DataTransferJob.java:96)
	at org.jkiss.dbeaver.tools.transfer.wizard.DataTransferJob.run(DataTransferJob.java:65)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:103)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)
Caused by: org.postgresql.util.PSQLException: ERROR: current transaction is aborted, commands ignored until end of transaction block
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2284)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2003)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:200)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:424)
	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:321)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:313)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.execute(JDBCStatementImpl.java:345)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:138)
	... 9 more

**Comment by serge-rider** [comment]
You can disable this on second wizard page:
![image](https://cloud.githubusercontent.com/assets/6398845/21849013/9ab9753e-d814-11e6-88f0-2d338324d72a.png)

Row count is needed only for export progress visualization.

**Comment by alextakitani** [comment]
Perfect! Thank you so much!

**Comment by oulenz** [comment]
I also just ran into this, basically every time I tried to order by some column. The problem is that the error messages are wrong. For example...

```sql
select
	table_schema
	,table_name
	,column_name

from	information_schema.columns

order by
	table_schema
	,table_name
	,column_name
```

...gives the error message `ERROR: column "columns.table_schema" must appear in the GROUP BY clause or be used in an aggregate function`. With different queries I get errors about not being allowed to group by aggregate functions and about certain aliases not being columns.

So it would be great if this could be fixed at some point down the line, and if `select row count` could be disabled by default until then. :-)

**Comment by serge-rider** [comment]
This problem is already resolved in 4.0 (in a more general fashion). DBeaver will rollback current transaction if row count will fail.

Generally converting any custom SQL into SELECT COUNT(*) is tricky. So I don't want to spend too much time on this - as we already have a workaround.
